src/Tearing/TearingEngine.inl

```diff
@@ -739,54 +726,47 @@ inline Index TearingEngine<DataTypes>::computeVertexByUnweightedAverage()
             {
                 // Add the triangle to ValidTrianglesAround
                 ValidTrianglesAround.push_back(trianglesAround[tri]);
-               
             }
         }
 
         
-        Real averageStress = 0.0;
-       
+        Real averageStress = 0.0;       
         for (unsigned int triID = 0 ; triID < ValidTrianglesAround.size() ; triID++)
         {
             const TriangleFEMInformation& tFEMinfo = triangleFEMInf[triID];
 
             if (tFEMinfo.area)
-             averageStress += fabs(tFEMinfo.maxStress);
+                averageStress += fabs(tFEMinfo.maxStress);
             
         }
 
         StressPerVertex[i] = averageStress;
-
     }
+
     Index k = (StressPerVertex[0] > StressPerVertex[1]) ? 0 : 1;
     k = (StressPerVertex[k] > StressPerVertex[2]) ? k : 2;
 
-    return(triangles[m_maxStressTriangleIndex][k]);
-   
+    return selectedTriangle[k];
 }
 
+
 template<class DataTypes>
-inline Index TearingEngine<DataTypes>::computeVertexByInverseDistance_WeightedAverage()
+inline Index TearingEngine<DataTypes>::computeVertexByInverseDistance_WeightedAverage(const Triangle& selectedTriangle)
 {
     helper::ReadAccessor< Data<VecTriangleFEMInformation> > triangleFEMInf(m_triangularFEM->triangleInfo);
 
     const auto& triangles = m_topology->getTriangles();
 
     constexpr size_t numVertices = 3;
-    sofa::type::vector<Index> VertexIndicies(numVertices);
     sofa::type::vector<Real>  StressPerVertex(numVertices);
-    VertexIndicies[0] = (triangles[m_maxStressTriangleIndex])[0];
-    VertexIndicies[1] = (triangles[m_maxStressTriangleIndex])[1];
-    VertexIndicies[2] = (triangles[m_maxStressTriangleIndex])[2];
-
 
     helper::WriteAccessor< Data<vector<Index>> >triangleToSkip(d_trianglesToIgnore);
 
     for (unsigned int i = 0; i < numVertices; i++)
     {
         std::vector<double> weight;
-        sofa::type::vector<Index> ValidTrianglesAround;
-        calculate_inverse_distance_weights(weight, VertexIndicies[i], ValidTrianglesAround);
+        sofa::type::vector<TriangleID> ValidTrianglesAround;
```

Comment:
It is more to makes the code clearer, so you directly know that we store Triangle indices in this vector and not vertex indices for example. 
Then it is also for code coherence in case one day we decide that Triangle indices are now a different type then other Indices (although it is almost certain that this will never happen)